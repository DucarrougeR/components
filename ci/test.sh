#!/usr/bin/env bash
yarn build
yarn test
