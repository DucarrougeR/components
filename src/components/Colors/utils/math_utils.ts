/**
 * Utility to convert a numerical value to a percent given the value and a max value.
 */
export const toPerc = (value: number, maxValue: number) =>
  Math.round((value / maxValue) * 100)
