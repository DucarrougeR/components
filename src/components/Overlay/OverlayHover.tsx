import * as React from 'react'
import { Popper } from 'react-popper'
import { CustomizableModalAttributes } from '../Modal'
import { OverlayProps } from './Overlay'

export interface OverlayHoverProps extends OverlayProps {
  setSurfaceRef?: (ref: HTMLElement | null) => void
  onMouseOut?: (event: React.MouseEvent) => void
}

export const OverlayHover: React.SFC<OverlayHoverProps> = ({
  children,
  isOpen,
  setSurfaceRef,
  ...props
}) => {
  const triggerRef =
    props.triggerRef && props.triggerRef.current
      ? props.triggerRef.current
      : undefined

  return isOpen ? (
    <Popper
      positionFixed
      innerRef={setSurfaceRef}
      placement={props.placement}
      referenceElement={triggerRef}
    >
      {({ ref, style, arrowProps, placement }) =>
        children({
          arrowProps,
          eventHandlers: { onMouseOut: props.onMouseOut },
          placement,
          ref,
          style: {
            ...style,
            zIndex: CustomizableModalAttributes.zIndex,
          },
        })
      }
    </Popper>
  ) : null
}
