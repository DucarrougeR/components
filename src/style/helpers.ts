import { css } from '../styled_components'
import { ThemedProps } from '../types'

export const reset = <P>(props: ThemedProps<P>) => props.theme.reset()

export const truncate = (doTruncate: boolean) => {
  if (doTruncate) {
    return css`
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    `
  }
  return ``
}
