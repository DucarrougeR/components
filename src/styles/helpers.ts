import { ThemedProps } from '../types'

export const reset = <P>(props: ThemedProps<P>) => props.theme.reset()
