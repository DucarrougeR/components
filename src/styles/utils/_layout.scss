// Returns a value from the $spacing list
@function spacing($key) {
  @return map-lookup($spacing, $key, '$spacing');
}
