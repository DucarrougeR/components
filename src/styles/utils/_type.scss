// Returns a value from the $test-size list
@function font-size($key) {
  @return map-lookup($text-size, $key, '$text-size');
 }

 // Returns a value from the $line-height list
 @function line-height($key) {
   @return map-lookup($line-height, $key, '$line-height');
 }

 @mixin fontSize($size) {
  font-size: $size;
  font-size: pxToRem($size);
}
