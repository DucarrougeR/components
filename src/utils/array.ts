/**
 * Returns an array of length `size` filled with zeros.
 * Useful for looping constructs where a length is known, but no array is provided.
 */

export const sizedArray = (size: number) => Array(size).fill(0)
