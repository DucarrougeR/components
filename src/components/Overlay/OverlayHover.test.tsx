import 'jest-styled-components'
import * as React from 'react'
import { mountWithTheme } from '../../../test/utils/create_with_theme'
import { assertSnapshot } from '../../../test/utils/snapshot'
import { SimpleContentSFC } from './overlay.test.helpers'
import { OverlayHover } from './OverlayHover'

interface OverlayTestProps {
  isOpen?: boolean
}

const SimpleOverlayHover: React.FC<OverlayTestProps> = ({ ...props }) => (
  <OverlayHover {...props}>{SimpleContentSFC}</OverlayHover>
)

describe('Overlay', () => {
  test('Generates a simple instance', () => {
    assertSnapshot(<SimpleOverlayHover isOpen />)
  })

  test('Generates a simple instanced, closed', () => {
    assertSnapshot(<SimpleOverlayHover />)
  })

  test('Contains content', () => {
    const content = <div>simple content</div>
    const contentFC = () => content
    const overlay = mountWithTheme(
      <OverlayHover isOpen>{contentFC}</OverlayHover>
    )

    expect(overlay.contains(content)).toBeTruthy()
  })
})
