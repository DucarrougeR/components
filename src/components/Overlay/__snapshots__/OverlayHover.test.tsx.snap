// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Generates a simple instance 1`] = `
<div>
  simple content
</div>
`;

exports[`Overlay Generates a simple instanced, closed 1`] = `null`;
